apps/web/src/pages/templates/components/CreateWorkflowDropdown.tsx

```diff
@@ -0,0 +1,109 @@
+import { useState } from 'react';
+import { Skeleton } from '@mantine/core';
+import { IconName } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faFile } from '@fortawesome/free-regular-svg-icons';
+import { faDiagramNext } from '@fortawesome/free-solid-svg-icons';
+import styled from '@emotion/styled';
+
+import { Button, Dropdown, Popover } from '../../../design-system';
+import { PlusCircle } from '../../../design-system/icons';
+
+const WIDTH = 172;
+
+const ButtonStyled = styled(Button)`
+  width: ${WIDTH}px;
+`;
+
+const DropdownItemSkeleton = styled(Skeleton)`
+  margin-bottom: 4px;
+  width: 100%;
+  height: 42px;
+`;
+
+export const CreateWorkflowDropdown = ({
+  readonly,
+  blueprints,
+  isLoading,
+  allTemplatesDisabled,
+  onBlankWorkflowClick,
+  onTemplateClick,
+  onAllTemplatesClick,
+}: {
+  readonly?: boolean;
+  blueprints?: { id: string; name: string; description: string; iconName: IconName }[];
+  isLoading?: boolean;
+  allTemplatesDisabled?: boolean;
+  onBlankWorkflowClick: React.MouseEventHandler<HTMLButtonElement>;
+  onTemplateClick: (template: { id: string; name: string; description: string; iconName: IconName }) => void;
+  onAllTemplatesClick: React.MouseEventHandler<HTMLButtonElement>;
+}) => {
+  const [templateId, setTemplateId] = useState<string | null>(null);
+
+  return (
+    <Dropdown
+      position="bottom"
+      disabled={readonly}
+      withArrow={false}
+      width={WIDTH}
+      styles={{
+        itemLabel: { textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap' },
+      }}
+      control={
+        <div>
+          <ButtonStyled disabled={readonly} icon={<PlusCircle />} data-test-id="create-template-btn">
+            Create Workflow
+          </ButtonStyled>
+        </div>
```

Comment:
had to wrap it within the div, otherwise, the dropdown is not positioned appropriately